hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/HiveSyncTool.java

```diff
@@ -298,6 +298,22 @@ protected void syncHoodieTable(String tableName, boolean useRealtimeInputFormat,
     LOG.info("Sync complete for " + tableName);
   }
 
+  private boolean needToSyncAllPartitions(Option<String> lastCommitTimeSynced) {
+    if (!lastCommitTimeSynced.isPresent()) {
+      return true;
+    }
+    if (config.getBoolean(META_SYNC_USE_STATE_TRANSIENT_TIME)) {
+      // If we use state transient time to sync partitions and the last commit time synced is before latest archive time
+      // We need to fall back to list all partitions instead of load the whole archive timeline
+      Option<String> latestArchiveTime = syncClient.getLastArchiveTime();
```

Comment:
Maybe we need to scan some archive logs here, at least those created after `lastCommitTimeSynced`.  Otherwise, we need to sync all partitions every time we do the archive.

Reply:
I believe we can get rid of the config option `META_SYNC_USE_STATE_TRANSIENT_TIME` if we keep both the start time(instant time) and max completion time(transition time) in the HMS.

We can use the instant time to check the max version id, and the completion time for real sync progress. Then we can fix the 'hollow' instants that are missed, just like what I fix in commit: https://github.com/apache/hudi/pull/8611

Reply:
Thanks for your advice~Will take a look at that pr and see if we can get rid of introducing the config `META_SYNC_USE_STATE_TRANSIENT_TIME `

Reply:
@danny0405 Hi Danny, syncing both instant time and completion time to HMS will not help solve the archive issue. Assume we have an instant completed a long time after its instant time(for example 1 day), the archiver will archive it immediately. And if we just looking the active timelime to find the 'hollow' instants we will miss the instant just have been archived. To solve this problem we still need to know the max completion time in archive timeline anyway. What do you think?

Reply:
> the archiver will archive it immediately. And if we just looking the active timelime to find the 'hollow' instants we will miss the instant just have been archived

There is no good manner to resolve the problem at this stage for Hudi, because we have no effeicent scanning and filtering on the archived timeline, what we can address in the PR is the multi-writer scenarios, e.g., an instant which is inflight first whith following complete instants and then finishes, without the fix, it could be skipped.